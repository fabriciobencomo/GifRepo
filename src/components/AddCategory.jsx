import {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('')
  const onInputChange = (e) => {
    setInputValue(e.target.value)
  }
  const onSubmit = (e) => {
    e.preventDefault()
    if(inputValue.trim().length <= 1) return
    onNewCategory(inputValue)
    setInputValue('')
  }
  return (
    <>
      <form onSubmit={e => onSubmit(e)} aria-label='form'>
        <input type="text" placeholder='search' value={inputValue} onChange={e => onInputChange(e)}/>
      </form>
    </>
  )
}
AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
