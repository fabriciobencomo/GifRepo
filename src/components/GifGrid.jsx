import { GifItem } from "./GifItem"
import { useFetchGif } from "../hooks/useFetchGif"
import PropTypes from "prop-types"
export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGif( category )
  return (
    <>
      <h1>{ category }</h1>
      {
        isLoading && <h2>Cargando...</h2>
      }
      <div className="card-grid">{
          images.map(({title, id, url}) => ( 
            <GifItem key={id} url={url} title={title}/>
          ))
        }</div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
