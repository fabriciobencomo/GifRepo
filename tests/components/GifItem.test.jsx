import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('GifItem', () => { 
    const title = "Saitama"
    const url = "https://one_punch.jpg/"
    test('Debe Hacer match el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url}></GifItem>)
        expect(container).toMatchSnapshot()
    })
    test('Debe mostrar la imagen con la url y el alt indicado', () => { 
        render(<GifItem title={title} url={url}></GifItem>)
    //    expect( screen.getByRole('img').src ).toBe(url)
    //    expect( screen.getByRole('img').alt ).toBe(title)
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })
    test('Debe mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url}></GifItem>)
        expect(screen.getByText(title)).toBeTruthy(). 
    })
})