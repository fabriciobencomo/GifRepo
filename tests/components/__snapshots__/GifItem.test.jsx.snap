// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Debe Hacer match el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one_punch.jpg/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
